### PR TITLE
Hocs-4235 -'createStage' is using a slow query

### PR DESCRIPTION
**Pull request by nickmcmahon01**
This commit
1) Gets 'createStage' to use the simpler (quicker) repository method without the topics/correspondents/case data.
2) Puts the database code in a transaction as we're modifying two tables
3) Removes multiple calls to getCaseData when one will do.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java [hunk 1]
```diff
@@ -189,10 +189,14 @@ public String getCaseDataCaseRef(UUID caseUUID) {
     }
 
     public String getCaseDataField(UUID caseUUID, String key) {
-        log.debug("Looking up key {} for Case: {}", key, caseUUID);
-        Map<String, String> dataMap = getCaseData(caseUUID).getDataMap(objectMapper);
+        return getCaseDataField(getCaseData(caseUUID), key);
+    }
+
+    public String getCaseDataField(CaseData caseData, String key) {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java:195]: Is this abstraction actually necessary, Line 308 does the same `getCaseData` call. Worth just moving this within the UUID for now. We can always add this back in the future...

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/CreateStageResponse.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
     @JsonProperty("uuid")
     private final UUID uuid;
 
-    public static CreateStageResponse from(StageWithCaseData stage) {
+    public static CreateStageResponse from(BaseStage stage) {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/dto/CreateStageResponse.java:18]: This response literally only needs a UUID, I see why you haven't done it to minimize the diff, but feels silly chucking the whole object around...

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java [hunk 3]
```diff
@@ -302,16 +306,21 @@ private CaseData createCaseBaseData(String caseType, Map<String, String> data, L
     }
 
     public void updateCaseData(UUID caseUUID, UUID stageUUID, Map<String, String> data) {
-        log.debug("Updating data for Case: {}", caseUUID);
+        if (data != null) {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java:309]: If data is null, we don't log, ideal?

**Review by LiamMacP — APPROVED** [review]

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java [hunk 4]
```diff
@@ -302,16 +306,21 @@ private CaseData createCaseBaseData(String caseType, Map<String, String> data, L
...
+            updateCaseData(getCaseData(caseUUID), stageUUID, data);
+        }
+    }
+
+    public void updateCaseData(CaseData caseData, UUID stageUUID, Map<String, String> data) {
+        log.debug("Updating data for Case: {}", caseData.getUuid());
         if (data != null) {
```
[the remark this answers is not in the document]
Comment (nickmcmahon01) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java:316]: I did that and put it back. I'll do it then :)

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java [hunk 5]
```diff
@@ -126,50 +126,45 @@ public CaseData getCase(UUID caseUUID) {
...
             throw new ApplicationExceptions.EntityNotFoundException(String.format("Case: %s, not found!", reference), CASE_NOT_FOUND);
         }
+        log.info("Got Case by reference: {}", caseData.getUuid(), value(EVENT, CASE_RETRIEVED));
+        return caseData;
     }
 
     private CaseData getAnyCaseData(UUID caseUUID) {
         log.debug("Getting any Case: {}", caseUUID);
         CaseData caseData = caseDataRepository.findAnyByUuid(caseUUID);
         if (caseData != null) {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java:162]: This should be `==`?

File: src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java (hunk 3 above)
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java:304]: Could we do the same here for the `if == null`?

**Pull request merged**